### PR TITLE
[Clang][RISCV] Use Decl for checkRVVTypeSupport

### PR DESCRIPTION
**Pull request by eopXD**
Using ValueDecl will cause error for OpenMP. Decl should do the work.

### CODE REVIEW

**Comment by alexey-bataev** [comment]
Need to add the test

File: clang/test/Sema/riscv-vector-with-openmp.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v \
+// RUN:   -disable-O0-optnone -o - -fsyntax-only -fopenmp %s -verify 
+// REQUIRES: riscv-registered-target
+// expected-no-diagnostics
+#include <riscv_vector.h>
```
Comment (alexey-bataev) [inline clang/test/Sema/riscv-vector-with-openmp.c:5]: No includes of system headers are allowed here, declare the type manually.

Comment (jrtc27) [reply clang/test/Sema/riscv-vector-with-openmp.c:5]: This is shipped by Clang not the system

File: clang/test/Sema/riscv-vector-with-openmp.c [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v \
+// RUN:   -disable-O0-optnone -o - -fsyntax-only -fopenmp %s -verify
```
Comment (jrtc27) [inline clang/test/Sema/riscv-vector-with-openmp.c:2]: -disable-O0-optnone is pointless unless you're generating code *and* piping it to opt, but you're doing neither. Similarly -o - is pointless with -fsyntax-only -verify given you're not producing output (except from -verify itself directly to stdio).

File: clang/test/Sema/riscv-vector-with-openmp.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v \
+// RUN:   -disable-O0-optnone -o - -fsyntax-only -fopenmp %s -verify 
+// REQUIRES: riscv-registered-target
+// expected-no-diagnostics
```
Comment (jrtc27) [inline clang/test/Sema/riscv-vector-with-openmp.c:4]:
```suggestion

// expected-no-diagnostics

```

**Review by jrtc27 — CHANGES_REQUESTED** [review]

File: clang/test/Sema/riscv-vector-with-openmp.c [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v -fopenmp %s
```
Comment (alexey-bataev) [inline clang/test/Sema/riscv-vector-with-openmp.c:1]: Could you move the test to OpenMP directory?

Comment (jrtc27) [inline clang/test/Sema/riscv-vector-with-openmp.c:1]: Commenting here rather than the commit: What happened to -fsyntax-only -verify?

**Review by alexey-bataev — APPROVED** [review]
LG

**Comment by eopXD** [comment]
@jrtc27 Anymore comments before merging this?

**Pull request merged**